packages/next/server/load-components.ts

```diff
@@ -94,16 +97,28 @@ export async function loadComponents(
 
   const DocumentMod = await requirePage('/_document', distDir, serverless)
   const AppMod = await requirePage('/_app', distDir, serverless)
-  const ComponentMod = await requirePage(pathname, distDir, serverless)
+  const ComponentMod = await requirePage(
+    pathname,
+    distDir,
+    serverless,
+    isFlight
+  )
 
-  const [buildManifest, reactLoadableManifest, Component, Document, App] =
-    await Promise.all([
-      require(join(distDir, BUILD_MANIFEST)),
-      require(join(distDir, REACT_LOADABLE_MANIFEST)),
-      interopDefault(ComponentMod),
-      interopDefault(DocumentMod),
-      interopDefault(AppMod),
-    ])
+  const [
+    buildManifest,
+    reactLoadableManifest,
+    reactFlightManifest,
+    Component,
+    Document,
+    App,
+  ] = await Promise.all([
+    require(join(distDir, BUILD_MANIFEST)),
+    require(join(distDir, REACT_LOADABLE_MANIFEST)),
+    isFlight ? require(join(distDir, REACT_FLIGHT_MANIFEST)) : null,
+    interopDefault(ComponentMod),
+    interopDefault(DocumentMod),
+    interopDefault(AppMod),
+  ])
```

Comment:
Is this `Promise.all` actually needed 👀  These don't look like promises  👀

Reply:
Fixed in #29494.